src/sentry/migrations/0912_split_discover_dataset_dashboards_self_hosted.py

```diff
@@ -0,0 +1,83 @@
+# Generated by Django 5.2.1 on 2025-05-20 17:45
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import Q
+
+from sentry.discover.dashboard_widget_split import _get_and_save_split_decision_for_dashboard_widget
+from sentry.models.dashboard_widget import DashboardWidgetTypes, DatasetSourcesTypes
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def split_discover_dataset_dashboards_self_hosted(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    DashboardWidgetQuery = apps.get_model("sentry", "DashboardWidgetQuery")
+    catch_all_unsplit_widgets = Q(
+        widget__widget_type=DashboardWidgetTypes.DISCOVER,
+    ) & ~Q(
+        widget__discover_widget_split__in=[
+            DashboardWidgetTypes.ERROR_EVENTS,
+            DashboardWidgetTypes.TRANSACTION_LIKE,
+        ]
+    )
+
+    queryset = DashboardWidgetQuery.objects.filter(
+        catch_all_unsplit_widgets,
+    ).select_related("widget__dashboard__organization")
+
+    for widget_query in RangeQuerySetWrapperWithProgressBar(queryset):
+        try:
+            _get_and_save_split_decision_for_dashboard_widget(widget_query, dry_run=False)
+        except Exception:
+            widget_query.widget.widget_type = DashboardWidgetTypes.ERROR_EVENTS
+            widget_query.widget.discover_widget_split = DashboardWidgetTypes.ERROR_EVENTS
+            widget_query.widget.dataset_source = DatasetSourcesTypes.UNKNOWN.value
+            widget_query.widget.save()
+
+
+def reverse_split_discover_dataset_dashboards_self_hosted(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    DashboardWidgetQuery = apps.get_model("sentry", "DashboardWidgetQuery")
+
+    queryset = DashboardWidgetQuery.objects.filter(
+        widget__widget_type=DashboardWidgetTypes.ERROR_EVENTS,
+        widget__dataset_source=DatasetSourcesTypes.UNKNOWN.value,
+    ).select_related("widget__dashboard__organization")
```

Comment:
Don't need the select related here. Also, this revert is missing transaction widgets:
```
    queryset = DashboardWidgetQuery.objects.filter(
        widget__discover_widget_split__in=[DashboardWidgetTypes.ERROR_EVENTS, DashboardWidgetTypes.TRANSACTION_LIKE],
    )
```